kmip_mgmt_adapters.go

```diff
@@ -0,0 +1,157 @@
+package kp
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+const (
+	KMIPAdapterPath = "kmip_adapters"
+	kmipAdapterType = "application/vnd.ibm.kms.kmip_adapter+json"
+)
+
+type KMIPAdapter struct {
+	ID          string      `json:"id,omitempty"`
+	Profile     string      `json:"profile,omitempty"`
+	ProfileData kmipProfile `json:"profile_data,omitempty"`
+	Name        string      `json:"name,omitempty"`
+	Description string      `json:"description"`
+	CreatedBy   string      `json:"created_by,omitempty"`
+	CreatedAt   *time.Time  `json:"created_at,omitempty"`
+	UpdatedBy   string      `json:"updated_by,omitempty"`
+	UpdatedAt   *time.Time  `json:"updated_at,omitempty"`
+}
+
+type KMIPAdapters struct {
+	Metadata KeysMetadata  `json:"metadata"`
+	Adapters []KMIPAdapter `json:"resources"`
+}
+
+type kmipProfile interface {
+	Type() string
+}
```

Comment:
I'm wondering whether this should remain private, users receiving a `KMIPAdapter` might find `ProfileData` to be an unknown type 🤔

Reply:
Simplified this part of code to just be a map[string]string